src/checkout-buttons/strategies/apple-pay/apple-pay-button-initialize-options.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * A set of options that are required to initialize ApplePay in cart.
+ *
+ * When ApplePay is initialized, an ApplePay button will be inserted into the
+ * DOM. When a customer clicks on it, it will trigger Apple sheet.
+ */
+ export default interface ApplePayButtonInitializeOptions {
+    /**
+     * The class name of the ApplePay button style.
+     */
+    buttonClassName?: string;
+
+    /**
+     * A callback that gets called when a payment is successfully completed.
+     */
+    onPaymentAuthorize(): void;
+
+    /**
+     * A callback that gets called if unable to initialize the widget or select
+     * one of the address options provided by the widget.
+     *
+     * @param error - The error object describing the failure.
+     */
+    onError?(error?: Error): void;
```

Comment:
Do we still need this? Since we are not using this in bcapp?

Reply:
Thought we need it. It appears on almost every `catch` inside `apple-pay-button-strategy.ts` to support the error handling.

Reply:
Yes, I think it's a good one to have as part of contract as well. But from the bcapp PR thought we had some difficulties implementing it?

Reply:
We are just lacking ideas for implementing `onError`.
With `onError`,  a merchant could display a message after ApplePay error or other tips to guide a shopper through the checkout.